### PR TITLE
Improve/fix RCPV macro support and docs

### PR DESCRIPTION
**Pull request by demerphq**
scope.c - improved RCPV support, SAVERCPV SAVEFREERCPV

I misnamed some of the RCPV stuff when it was introduced. The macro which I called SAVERCPVFREE should have been called SAVEFREERCPV, and there should also have been a SAVERCPV macro.

There also should have been RCPV_REFCNT_dec() and RCPV_REFCNT_inc() macros.

There was also missing documentation. This patch should fix all of these issues. Since this functionality has never been released in a production perl it should be fine to do these renames, nothing out there should use these macros yet.

I noticed these oversights while fixing Scope::Upper, see also:

https://github.com/Perl/perl5/issues/20861
https://rt.cpan.org/Ticket/Display.html?id=146897

### CODE REVIEW

File: pod/perlguts.pod [hunk 1]
```diff
@@ -1889,6 +1889,20 @@ and back.
...
+
+This macro arranges to restore the value of a C<char *> which was
+allocated with a call to C<rcpv_new()> to its previous state when the
+current pseudo block is completed. The pointer stored in C<pv> at the
+time of the call will be refcount incremented and stored on the save
+stack. Later when the current pseudoblock is completed the value stored
+in C<&rcpv> will be refcount decremented, and the previous value
+restored from the savestack and also be refcount decremented.
+
+This is the C<RCPV> equivalent of C<SAVEFREESV()>.
```
Comment (leonerd) [inline pod/perlguts.pod]: Did you mean to put this line in the docs for `SAVEFREERCPV()`? It doesn't seem to make sense here

**Review by leonerd — COMMENTED** [review]
Overall seems good but one question

File: pod/perlguts.pod (hunk 1 above)
Comment (demerphq) [reply pod/perlguts.pod]: Ah, good catch. Thanks. (The naming conventions and layout for these docs is really confusing and irritating.)

Comment (demerphq) [reply pod/perlguts.pod]: Anyway, the latest patch should fix this, and add docs for what that should have said SAVEGENERICSV(). (Like I said the naming convention for that is sucky.)

**Review by tonycoz — APPROVED** [review]

**Pull request merged**